examples/wasi-http-rust-async/src/lib.rs

```diff
@@ -0,0 +1,307 @@
+wit_bindgen::generate!({
+    world: "wasi:http/proxy",
+    path: "wit",
+    exports: {
+        "wasi:http/incoming-handler": Component
+    }
+});
+
+use {
+    self::{
+        exports::wasi::http::incoming_handler::Guest as IncomingHandler,
+        wasi::{
+            http::{
+                outgoing_handler,
+                types::{
+                    self, Fields, IncomingBody, IncomingRequest, IncomingResponse, Method,
+                    OutgoingBody, OutgoingRequest, OutgoingResponse, ResponseOutparam, Scheme,
+                },
+            },
+            io::streams::StreamError,
+        },
+    },
+    anyhow::{anyhow, bail, Error, Result},
+    futures::{future, sink, stream, FutureExt, Sink, SinkExt, Stream, StreamExt, TryStreamExt},
+    sha2::{Digest, Sha256},
+    std::{cell::RefCell, future::Future, rc::Rc, str, task::Poll},
+    url::Url,
+    wakers::Wakers,
+};
+
+mod wakers;
+
+const READ_SIZE: u64 = 16 * 1024;
+const MAX_CONCURRENCY: usize = 16;
+
+struct Component;
+
+impl IncomingHandler for Component {
+    fn handle(request: IncomingRequest, response_out: ResponseOutparam) {
```

Comment:
The handling here is definitely not as straight-forward as before. What do we plan to do to make this more approachable?

Reply:
Yeah, doing async without an SDK is going to be a bit ugly until Preview 3's first-class async is available.  My plan in the meantime is to provide wasi-http support in the Rust, JS, and Python SDKs that hides details like these.  Not sure how the Go SDK is going to look -- I'm leaving that up to the TinyGo/Go folks to figure out.